### PR TITLE
Improve interface and examples

### PR DESCRIPTION
**Pull request by steveklabnik**
Instead of providing a &mut Request, provide a RequestBuilder.

### CODE REVIEW

File: examples/simple-server.rs [hunk 1]
```diff
@@ -1,14 +1,16 @@
 extern crate simple_server;
+extern crate http;
```
Comment (ashleygwilliams) [inline examples/simple-server.rs]: i would prefer we don't require "sibling" or "peer" deps as they are hard to discover. i think this library should re-export them.

File: src/lib.rs [hunk 2]
```diff
@@ -18,13 +18,13 @@ use std::net::{TcpListener, TcpStream};
 /// |-----------|--------------------------------------------|---------------------------------------------------------------------|
 /// | `handler` | `fn(Request<&[u8]>, &mut Response<&[u8]>)` | This function uses Types that are re-exported from the `http` crate |
 pub struct Server {
-    handler: fn(Request<&[u8]>, &mut Response<&[u8]>),
+    handler: fn(Request<&[u8]>, http::response::Builder) -> Response<&[u8]>,
```
Comment (ashleygwilliams) [inline src/lib.rs]: perhaps we can do `use http::response::Builder as ResponseBuilder;` here like in the example

File: tests/simple-server.rs [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 #[test]
 fn test_new() {
-    fn handler(request: Request<&[u8]>, response: &mut Response<&[u8]>) {
-        *response.body_mut() = "Hello Rust!".as_bytes();
+    fn handler(_request: Request<&[u8]>, mut response_builder: ResponseBuilder) -> Response<&[u8]> {
+        response_builder.body("Hello Rust!".as_bytes()).unwrap()
     };
 
-    let server = Server::new(handler);
+    let _server = Server::new(handler);
```
Comment (ashleygwilliams) [inline tests/simple-server.rs]: we could probably just not assign this

File: src/lib.rs (hunk 2 above)
Comment (steveklabnik) [inline src/lib.rs]: yeah i just messed this one up

**Pull request merged**